text/0005-application-formalization.md

```diff
@@ -0,0 +1,194 @@
+* Start Date: 10/09/2023
+* Contributors:
+  * [Craig McNally](cmcnally@ebsco.com)
+  * [Vince Bareau](vbareau@ebsco.com)
+* RFC PRs:
+  * PRELIMINARY REVIEW: https://github.com/folio-org/rfcs/pull/14
+  * DRAFT REFINEMENT: https://github.com/folio-org/rfcs/pull/22
+  * PUBLIC REVIEW: TBD
+  * FINAL REVIEW: TBD
+* Outcome: <!-- Leave this blank. Will eventually be either ACCEPTED or REJECTED -->
+
+# Application Formalization
+
+## Summary
+It is proposed that an Application be defined as the minimal but complete set of elements which together are intended to deliver a specific solution to Folio. The intention is to group together all the closely related components which will need to evolve together, but independently of other parts (i.e. other applications) of Folio.
+
+## Motivation
+The existing Folio project lacks a formal definition of an Application structure. A consequence is that Folio is stuck in a mode of monolithic releases which are ever increasing in size.  There have also been repeated complaints that the setup and management of Folio can be difficult due to the number modules involved.
+
+## Scope
+* Application composition
+* Application versioning
+* Application dependencies
+* Application definition/descriptors
+* Backward compatibility with Folio instances not adopting Applications
+* Application management is ***out of scope***
+* Application Stores and Marketplaces are ***out of scope***
+* Application-specific UI bundles is ***out of scope***
+* How movement of modules between Applications will work is ***out of scope***
+* How bounded contexts and cross-module database access could be introduced is ***out of scope***
+
+## Detailed Explanation/Design
+
+### Terminology
+* Application:  The minimal but complete set of elements which together are intended to deliver a specific solution to Folio.
```

Comment:
The use of "complete" has caused some confusion in earlier conversations in multiple venues, as people assume this means no cross-application use of interfaces. That's cleared up in the example below, but perhaps useful to state explicitly that an application may depend on other applications for data or processes that it makes use of but does not bear primary responsibility for.

Reply:
Do you have a suggested (word or phrasing) which would make it more clear?

Reply:
Hmm, maybe the end of that comment works okay as an additional sentence:
"Note that an application may depend on other applications for data or processes that it makes use of, but does not bear primary responsibility for."

I'm trying to convey that an application bears primary responsibility for some set of functionality and data in order to deliver whatever solution, that's the scope of the application, but will often depend on other applications for supporting data and processes.

Reply:
It's almost like the problem isn't with the definition itself, but rather the position of the terminology in the document... People will read this first and come to conclusions prior to clarifying text/examples later in the document.

Reply:
I personally think keeping the terminology concise is the way to go here.  Does a definition of application really require delving into the dependency aspect?  Should terminology/definitions have footnotes?  My gut reaction to both no, but I'd like to hear from others on this.

Reply:
@todolson any thoughts on my response above?

Reply:
I think keep it concise and see whether it generates confusion in the next phase.